src/components/PostDetailed.js

```diff
@@ -1,22 +1,52 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Card, Button } from 'antd';
 import { Link } from 'react-router-dom';
 
-const PostDetailed = ({ post, handleDelete }) => (
-  <div>
-    {post && (
-      <Card title={post.title} style={{ width: 300 }}>
-        <p>
-          <b>Author:</b> {post.author}
-        </p>
-        <p>
-          <b>Text:</b> {post.text}
-        </p>
-        <Link to={'/edit/' + post.id}>Edit</Link>
-        <Button onClick={() => handleDelete(post.id)}>Delete</Button>
-      </Card>
-    )}
-  </div>
-);
+import { connect } from 'react-redux';
+import { selectPost, deletePost } from 'redux/modules/posts';
 
-export default PostDetailed;
+class PostDetailed extends Component {
+  constructor(props) {
```

Comment:
No need for constructor, we can define methods via arrow functions so they will get bound to `this`. Also use destructuring to reduce amount of boilerplate code.

Eg:

```

deletePost = () => {
  const { deletePost, history, match } = this.props;
  deletePost(match.params.id);
  history.push('/');
}

```

Reply:
Also what happens if deleting a post fails?